kernel/src/syscall/pidfd_open.rs

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::sync::atomic::Ordering;
+
+use crate::{
+    fs::{file_table::FdFlags, utils::StatusFlags},
+    prelude::*,
+    process::{posix_thread::PID_MAX, process_table, Pid, PidFile},
+    syscall::SyscallReturn,
+};
+
+pub fn sys_pidfd_open(pid: Pid, flags: u32, ctx: &Context) -> Result<SyscallReturn> {
+    let is_nonblocking = {
+        let flags = PidfdFlags::from_bits(flags)
+            .ok_or_else(|| Error::with_message(Errno::EINVAL, "invalid flags"))?;
+        debug!("pid = {}, flags = {:?}", pid, flags);
+        flags.contains(PidfdFlags::PIDFD_NONBLOCK)
+    };
+
+    if pid > PID_MAX.load(Ordering::Acquire) {
```

Comment:
I guess this is unrelated. PID can be larger than `/proc/sys/kernel/pid_max`, otherwise, what happens if you reduce `pid_max` to a lower value.

This may be something like:
https://github.com/asterinas/asterinas/blob/00e6905d93b19784cdd04585b1360f95b3c8540f/kernel/src/syscall/setpgid.rs#L15-L17